lldb/source/API/SBProcess.cpp

```diff
@@ -1222,7 +1223,17 @@ lldb::SBError SBProcess::SaveCore(const char *file_name) {
 lldb::SBError SBProcess::SaveCore(const char *file_name,
                                   const char *flavor,
                                   SaveCoreStyle core_style) {
-  LLDB_INSTRUMENT_VA(this, file_name, flavor, core_style);
+  SBCoreDumpOptions options(file_name);
```

Comment:
I would default construct the SBCoreDumpOptions and then call:
```
options.SetOutputFile(file_name);
```
If a user wants to make a command line tool that uses this API, they will need to default construct a `SBCoreDumpOptions` object and fill things in as needed. If we require a filename up front, then that ruins the ability to use the `SBCoreDumpOptions` class effectively.